int32 a
---
int32 ans